src/DI/MonologExtension.php

```diff
@@ -101,43 +103,45 @@ public function loadConfiguration()
 		$this->loadProcessors($config);
 	}
 
-	protected function loadHandlers(array $config)
+	protected function loadHandlers(array $config): void
 	{
 		$builder = $this->getContainerBuilder();
 
 		foreach ($config['handlers'] as $handlerName => $implementation) {
-			Compiler::loadDefinitions($builder, [
-				$serviceName = $this->prefix('handler.' . $handlerName) => $implementation,
-			]);
+
+			$sd = new ServiceDefinition();
+			$sd->setFactory($implementation)->setAutowired(FALSE);
+			$serviceName = $this->prefix('handler.' . $handlerName);
+			$builder->addDefinition($serviceName, $sd);
 
 			$builder->getDefinition($serviceName)
 				->addTag(self::TAG_HANDLER)
 				->addTag(self::TAG_PRIORITY, is_numeric($handlerName) ? $handlerName : 0);
```

Comment:
The block above looks like it can be simplified.

- `new ServiceDefinition()` should be dropped in favor of `$builder->addDefinition(...)`
- `$sd` should not be necessary, just use single chain of methods. If it is necessary rename to `$definition`
- `$builder->getDefinition(...)` should not be needed I think

Reply:
You're right of course. Thank you. Will update the PR.

Reply:
Done. Commit https://github.com/haltuf/Monolog/commit/92beb8e009618ff22d3f14cd78edeb32b44ae89e pushed to @JanGalek's repository.